Re-architect format.py in swift (#2022)
apple/swift-syntax: CONTRIBUTING.md

swift-syntax is formatted using [swift-format](http://github.com/apple/swift-format) to ensure a consistent style.

> To format your changes run `format.swift` at the root of this repository from the terminal as `./format.swift`. It will build a local copy of swift-format from the `main` branch and format the repository. Since it is building a release version of `swift-format`, the first run will take few minutes. Subsequent runs take less than 2 seconds.

Reviewer: ```suggestion
To format your changes, run `format.swift` at the root of this repository from the terminal as eg. `./format.swift`. It will build a local copy of swift-format from the `main` branch and format all files. Since it is building a release version of `swift-format`, the first run will take few minutes. Subsequent runs are much faster (a couple seconds).
```

Generated source code is not formatted to make it easier to spot changes when re-running code generation.

> > 💡 Tip: You can add a git hook to ensure all commits to the swift-syntax repository are correctly formatted. 

Reviewer: ```suggestion
> [!NOTE]
> You can add a git hook to ensure all commits to the swift-syntax repository are correctly formatted. 
```

> > 1. Save the following contentes to `swift-syntax/.git/hooks/pre-commit`

Reviewer 2: Is there a typo in `contentes`?

> ```bash

> > #!/bin/bash

Reviewer: ```suggestion
> #!/usr/bin/env bash
```

> set -e
> SOURCE_DIR=$(realpath "$(dirname $0)/../..")
> "$SOURCE_DIR/format.swift" --lint
> ```
> 2. Mark the file as executable: `chmod a+x swift-syntax/.git/hooks/pre-commit`

> > 3. If you have any global git hooks installed, be sure to call them at the end of the script: `path/to/global/hooks/pre-commit "$@" 

Reviewer: ```suggestion
> 3. If you have any global git hooks installed, be sure to call them at the end of the script with `/path/to/global/hooks/pre-commit "$@"`
```
